common/src/Microsoft.Azure.IIoT.Validators.JsonSchemaDotNet/tests/Default/SchemaValidationTests.cs

```diff
@@ -8,24 +8,64 @@ namespace Microsoft.Azure.IIoT.Validators.JsonSchemaDotNet.Tests.Default {
     using System.IO;
     using System.Linq;
     using System.Text;
+    using System.Text.Json;
     using Xunit;
 
     /// <summary>
-    /// These tests are designed to exercise the 
-    /// generated schema for configuration file 
-    /// validation. These tests are additionally 
-    /// found in the schema generator tool located
+    /// These tests are designed to exercise the generated schema for configuration file 
+    /// validation. These tests are additionally found in the schema generator tool located
     /// at: https://github.com/WilliamBerryiii/opcpublisherschemavalidator
+    /// 
+    /// The referenced schema file is a linked asset in the project file set to 
+    /// copy to the output build directory so that it can be easily referenced here.
     /// </summary>
     public class SchemaValidationTests {
 
+        // allow training commas in the test parsers as the main serializer for 
+        // this solution supports training commas. 
+        private static JsonDocumentOptions parseOptions = new JsonDocumentOptions {
+            AllowTrailingCommas = true,
+        };
+
         [Fact]
         public void EnsureTestConfigurationPassesSchemaValidation() {
-            using var schemaReader = new StreamReader("Storage/publishednodesschema.json");
+            using var schemaReader = new StreamReader("Default/publishednodesschema.json");
             var validator = new JsonSchemaDotNetSchemaValidator();
-            var results = validator.Validate(Encoding.UTF8.GetBytes(pn), new StringReader(schema));
+            var results = validator.Validate(Encoding.UTF8.GetBytes(testConfiguration), schemaReader);
             Assert.True(results.All(r => r.IsValid));
+        }
+
+        [Fact]
+        public void ByteStringNodeIdsThatAreNotByteStingsReturnErrors() {
+            using var schemaReader = new StreamReader("Default/publishednodesschema.json");
+            // break the id of the first Node to ensure an error is thrown
+            var alteredConfig = testConfiguration.Replace("i=1001", "b=12345");
+            var validator = new JsonSchemaDotNetSchemaValidator();
+            var results = validator.Validate(Encoding.UTF8.GetBytes(alteredConfig), schemaReader);
+            // ensure that we failed the regex NodeID check
+            Assert.True(results.Any(r => r.Message == "The string value was not a match for the indicated regular expression"));
```

Comment:
If we could use Assert.Equal instead that would benefit from the following:
1. Allows us specifically check the property we want to test for schema validation so there's no chance of test passing because some other property raised the same error incidentally.
2. When tests fail, Assert.Equal provides more insights on failure e.g. comparing expected and found strings in case.

Example:  Assert.Equal("The string value was not a match for the indicated regular expression",
                results.FirstOrDefault(r => r.InstanceLocation == "#/0/OpcNodes/0/Id").Message);

Reply:
Let's play with the basic vs detailed output and see what the result is of that test. My main concern is that if we go with Basic output, can we still see that we are not matching specific elements of the `anyof` schema. If basic just returns "no matching schema" without any details as to why it didn't match one of the sub-schema in the `anyof` then we should stick to detailed output ... once that's sorted, I'll pick a test method that returns the best error formulation and still feels idomatic.

Reply:
So it turns of that it does support anyof, I tried with the following to allow some level of nesting as well:
```
 "DataSetWriterGroup0": {
            "type": "object",
            "properties": {
              "DataSetWriterGroup1": {
                "type": "object",
                "properties": {
                  "DataSetWriterGroup2": {
                    "type": "object",
                    "properties": {
                      "DataSetWriterGroup3": {
                        "anyOf": [
                          {
                            "type": "string",
                            "maxLength": 5
                          },
                          {
                            "type": "number",
                            "minimum": 4
                          }
                        ]                        
                      }
                    }
                  }
                }
              }
            }
          },
```
Think we should change it to basic then for a simpler solution but still retaining validation results fidelity?

Reply:
Cool lets do it. Do you want to make the change (including updating the recursion) and I'll update the tests?

Reply:
Resolved in commits below